### DISCUSSION TITLE
What is the timestamp unit?

### DISCUSSION THREAD

**Discussion by jfarjona** [questdb/questdb#1337] [Ideas]
Hello:

I have been going through the documentation, but I cannot find the units of the timestamp when I use questDB embedded.
The method putTimeStamp(int index, long ts) has no javadoc.  Question is: ts should be in milliseconds? nanoseconds? microseconds?  I assume is epoch-based.  

Thanks,

Juan

**Comment by ideoma** [comment]
Microseconds

**Comment by bsmth** [comment]
Thanks for the question, Juan 🙌🏻 

Alongside ideoma's answer, there is a doc for working with timestamps here - https://questdb.io/docs/guides/working-with-timestamps-timezones

> The native timestamp format used by QuestDB is a Unix timestamp in microsecond resolution.
